build.sbt

```diff
@@ -27,14 +27,32 @@ ThisBuild / githubWorkflowBuild += WorkflowStep.Run(
   commands = "sbt --client docs/mdoc" :: Nil
 )
 
-lazy val root =
-  project
-    .in(file("."))
-    .enablePlugins(NoPublishPlugin)
-    .aggregate(chanterelle)
+lazy val jsSettings = Seq(
+  scalacOptions ++= {
+    val localSourcesPath = (LocalRootProject / baseDirectory).value.toURI
+    val remoteSourcesPath = s"https://raw.githubusercontent.com/arainko/chanterelle/${git.gitHeadCommit.value.get}/"
+    Seq(
+      s"-scalajs-mapSourceURI:$localSourcesPath->$remoteSourcesPath",
+      "-scalajs-genStaticForwardersForNonTopLevelObjects"
+    )
+  },
+  bspEnabled := false,
+  mimaPreviousArtifacts := Set()
+)
+
+lazy val nativeSettings = Seq(
+  scalacOptions ++= Seq("-P:scalanative:genStaticForwardersForNonTopLevelObjects"),
+  bspEnabled := false,
+  mimaPreviousArtifacts := Set()
+)
+
+lazy val root = tlCrossRootProject
+  .aggregate(chanterelleJVM, chanterelleJS, chanterelleNative)
+  .enablePlugins(NoPublishPlugin)
 
 lazy val chanterelle =
```

Comment:
I feel like yo can get away without expanding into multiple subprojects, going off of how I set it up in ducktape:
```scala
lazy val ducktape =
  crossProject(JVMPlatform, JSPlatform, NativePlatform)
    .crossType(CrossType.Pure)
    .enablePlugins(TypelevelMimaPlugin)
    .in(file("ducktape"))
    .settings(
      scalacOptions ++= List("-deprecation", "-Wunused:all", "-Ykind-projector:underscores", "-Xcheck-macros"),
      Test / scalacOptions --= List("-deprecation"),
      Test / scalacOptions ++= List("-Werror", "-Wconf:cat=deprecation:s"),
      libraryDependencies += "org.scalameta" %%% "munit" % "1.1.1" % Test
    )
    .nativeSettings(
      bspEnabled := false,
      tlMimaPreviousVersions := Set.empty
    )
    .jsSettings(bspEnabled := false)
```